client/src/app/home/home/transaction-card/services/transaction.service.ts

```diff
@@ -0,0 +1,57 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, tap } from 'rxjs';
+import { Transaction } from '../transaction.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TransactionService {
+  public transactions!: Transaction[];
+  public filteredTransactions!: Transaction[];
+  public isIncomeFilter: boolean = false;
+  public isExpenseFilter: boolean = false;
+
+  constructor(private http: HttpClient) {}
+
+  public getTransactions(accountId: string): Observable<any> {
+    return this.http
+      .get(`http://localhost:3000/accounts/${accountId}/transactions`)
+      .pipe(
+        tap({
+          next: (res: any) => {
+            this.transactions = res;
+            this.filteredTransactions = res;
+          },
+        })
+      );
+  }
+
+  public filterTransactions(type?: string): void {
+    if (!type) {
```

Comment:
do you really need to store filtered transactions? This function can return just return it

Reply:
It doesnt update the ui if it's not tied to a variable. But perhaps i just lack some knowledge of how to do it without the variable.